sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/objects.scala

```diff
@@ -96,3 +97,16 @@ object CombineTypedFilters extends Rule[LogicalPlan] {
     }
   }
 }
+
+/**
+ * Removes MapObjects when the following conditions are satisfied
+ *   1. Mapobject(e) where e is lambdavariable(), which means types for input output
```

Comment:
This comment is obscure. Can we improve it a bit?

For example, `MapObject(e)` is confusing. Shall we clearly say the lambdaFunction of `MapObject`?